python/ray/serve/tests/test_proxy_state.py

```diff
@@ -802,6 +809,52 @@ def check_worker_node_proxy_actor_is_removed():
     assert manager._proxy_states[HEAD_NODE_ID].status == ProxyStatus.HEALTHY
 
 
+@patch("ray.serve._private.proxy_state.PROXY_HEALTH_CHECK_PERIOD_S", 5)
+def test_proxy_state_reconcile_draining_success():
+    """Test that the proxy will remain DRAINING even if health check succeeds."""
+    timer = MockTimer(start_time=0)
+    # Start with HEALTHY state
+    proxy_state = _create_proxy_state(status=ProxyStatus.HEALTHY, timer=timer)
+    # Simulate health-checks passing
+    proxy_state._actor_proxy_wrapper.healthy = ProxyWrapperCallStatus.FINISHED_SUCCEED
+    # Simulate is_drained check returning false
+    proxy_state._actor_proxy_wrapper.drained = ProxyWrapperCallStatus.FINISHED_FAILED
+
+    for _ in range(10):
+        proxy_state.update(draining=True)
+        assert proxy_state.status == ProxyStatus.DRAINING
+        # Advance timer by 5 (to trigger new health-check, drain-check)
+        timer.advance(5)
+
+    # assert proxy_state._actor_proxy_wrapper.get_num_health_checks() == 10
+    # assert proxy_state._actor_proxy_wrapper.get_num_drain_checks() == 9
```

Comment:
intentionally commented out?

Reply:
+1 on those commented out code

Reply:
Yes, forgot to remove, asserting on these is currently meaningless unfortunately.

I'm planning to supersede this by https://github.com/ray-project/ray/pull/41722/files